autoprotocol/protocol.py

```diff
@@ -5137,6 +5246,9 @@ def provision(self, resource_id, dests, amounts=None, volumes=None):
 
         measurement_mode = self._identify_provision_mode(provision_amounts)
 
+        if informatics is not None:
+            informatics = self.check_informatics(informatics, dests)
```

Comment:
hmm ideally we should do this as part of every Instruction, for every Informatics subclass? Thinking out loud, it'll be nice to define the checks just once and for each instruction with an Informatics, we'll automatically go through the defined checks.

I.e. have a `validate` abstractmethod defined for the base Informatics class, have each child class implement that method. During Instruction instantiation, we call the `validate` method for each attached Informatics object.

Reply:
Ya... that was the idea initially to run validation in Instruction... but I couldn't find a way to run validation on wells (to make sure Informatics wells are included in the instruction wells) at that level...? So I moved it to Protocol...